packages/hardware-ledger/src/transformers/txOut.ts

```diff
@@ -56,27 +56,33 @@ const getScriptHex = (output: Serialization.TransactionOutput): HexBlob | null =
   return scriptRef.toCbor();
 };
 
-export const toTxOut: Transform<Cardano.TxOut, Ledger.TxOutput, LedgerTxTransformerContext> = (txOut, context) => {
+export const toTxOut: Transform<
+  { txOut: Cardano.TxOut; index: number; isCollateral: boolean },
+  Ledger.TxOutput,
+  LedgerTxTransformerContext
+> = (elem, context) => {
+  const { txOut, index, isCollateral } = elem;
   const output = Serialization.TransactionOutput.fromCore(txOut);
   const scriptHex = getScriptHex(output);
+  const format = isCollateral ? context?.collateralReturnFormat : context?.outputsFormat[index];
 
-  return context?.useBabbageOutputs
+  return format === Ledger.TxOutputFormat.MAP_BABBAGE
```

Comment:
Probably out of scope for this PR, but here I see some code repetition. I'd refactor in something like

```
return {
  amount: <...>,
  destination: <...>,
  format,
  tokenBundle,
  ...(format === Ledger.TxOutputFormat.MAP_BABBAGE ?
    { datum: <...>, referenceScriptHex: <...> } :
    { datumHashHex: <...> })
};
```

Reply:
done here @iccicci https://github.com/input-output-hk/cardano-js-sdk/pull/1618/commits/d00d80a567431a1ed734502745b8b9793fd465fc for some reason the type checker complain if I use the spread and not set the format specifically, so had to duplicate that bit

Reply:
Nice!